pkg/kubelet/status/status_manager.go

```diff
@@ -634,6 +646,15 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 		pod = newPod
 	}
 
+	// measure how long the status update took to propagate from generation to update on the server
+	var duration time.Duration
+	if status.at.IsZero() {
+		klog.V(3).InfoS("Pod had no status time set", "pod", klog.KObj(pod), "podUID", uid, "version", status.version)
+	} else {
+		duration = time.Now().Sub(status.at).Truncate(time.Millisecond)
+	}
+	metrics.PodStatusSyncDuration.WithLabelValues(strconv.Itoa(0)).Observe(duration.Seconds())
```

Comment:
Should we only observe a duration if status.at was non-zero?

Reply:
Good question, it's typically an indicator that something unexpected happened, but it would skew the numbers.  Let me review.

Reply:
I'll only observe if it's set, and then verify there are no wierd cases where a synthetic status is created that somehow escapes time (can't be 100% sure, but log observation in runs should be sufficient).